### PR TITLE
update to 1.8.0

### PR DESCRIPTION
**Pull request by izahn**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

@jerowe indicated in a personal email that she no longer wishes to maintain this package, and so I have added myself as a maintainer.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,42 +1,46 @@
...
 
 requirements:
   build:
-    - gcc  # [linux]
-    - llvm  # [osx]
-    - zlib 1.2.*
+    - {{ compiler('c') }}
+  host:
     - luarocks
+    - libgcc  # [linux]
```
Comment (xhochy) [inline recipe/meta.yaml]:
```suggestion
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,42 +1,46 @@
...
+    - zlib
     - lua
-
   run:
-    - libgcc  # [linux]
-    - zlib 1.2.*
     - luarocks
-    - lua
+    - libgcc  # [linux]
+    - zlib
```
Comment (xhochy) [inline recipe/meta.yaml]:
```suggestion
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,42 +1,46 @@
...
 
 package:
   name: {{ name }} 
-  version: {{version }}
+  version: {{ version }}
 
 source:
-  git_url: git://github.com/keplerproject/luafilesystem
+  url: https://github.com/keplerproject/luafilesystem/archive/refs/tags/v1_8_0.tar.gz
+  sha256: {{ sha256 }}
```
Comment (xhochy) [inline recipe/meta.yaml]:
```suggestion
  sha256: 16d17c788b8093f2047325343f5e9b74cccb1ea96001e45914a58bbae8932495
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,42 +1,46 @@
 {% set name = "lua-luafilesystem" %}
-{% set version = "1.6.3" %}
-{% set sha256 = "d58439c548433adcda98e695be53e526ba940a4b9c44fb9a05d92cd495cdd47f" %}
+{% set version = "1.8.0" %}
+{% set sha256 = "16d17c788b8093f2047325343f5e9b74cccb1ea96001e45914a58bbae8932495" %}
```
Comment (xhochy) [inline recipe/meta.yaml]:
```suggestion
```

**Review by xhochy — COMMENTED** [review]
Added some suggestion but happy to merge this

**Comment by izahn** [comment]
Thank you @xhochy

**Pull request merged**